### PR TITLE
DEVELOPMENT: instructions for a local registry option

### PR DESCRIPTION
**Pull request by vbatts**
Instructions for a local, non-gcr.io registry.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -33,13 +33,25 @@ $ git remote add ${USER} https://github.com/${USER}/build
...
 
+For a gcr.io project (like "myproject"), it will need to have Container
+Registry API enabled first.
 ```shell
 # You can put these definitions in .bashrc, so this is one-time setup.
 #
 # KO_DOCKER_REPO is the image repository where the images for build
 # will be pushed, including hostname.
-export KO_DOCKER_REPO=us.gcr.io/project
+export KO_DOCKER_REPO=us.gcr.io/myproject
```
Comment (jonjohnsonjr) [inline DEVELOPMENT.md]: Can you also `s/myproject/<myproject>/` so that this fails early during parsing instead of at authentication time?

Comment (vbatts) [reply DEVELOPMENT.md]: good thought. done.

**Comment by vbatts** [comment]
all green

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -33,13 +33,25 @@ $ git remote add ${USER} https://github.com/${USER}/build
...
 # KO_DOCKER_REPO is the image repository where the images for build
 # will be pushed, including hostname.
-export KO_DOCKER_REPO=us.gcr.io/project
+export KO_DOCKER_REPO=us.gcr.io/<myproject>
+```
+
+For a local registry (using insecure http://) it needs to be on localhost:
+```shell
+docker run -it -d -p 5000:5000 registry:2
+export KO_DOCKER_REPO=localhost:5000/<myproject>
```
[the remark this answers is not in the document]
Comment (jonjohnsonjr) [reply DEVELOPMENT.md]: It should work - the repository component is just the path.

If you just do `KO_DOCKER_REPO=localhost:5000`, it won't work (it's ambiguous), but `KO_DOCKER_REPO=localhost:5000/foo` should.

What errors are you seeing?

[the remark this answers is not in the document]
Comment (greghaynes) [reply DEVELOPMENT.md]: Ah its exactly what you mention, sorry :)

**Comment by ImJasonH** [comment]
@vbatts Are these instructions still up-to-date? If so, we can merge this. Sorry it took so long to get back to it.

**Pull request merged**